queue.rb

```diff
@@ -0,0 +1,33 @@
+class Queue
+
+	def initialize 
+    @queue = []
+	end
+
+  def enqueue(item)
+    @stack[@queue.length] = item
+  end
+
+  def dequeue
+
+		new_queue = []
+		i = 1
+
+		while i < @queue.length
```

Comment:
instead of creating a new queue, shift the values in the current queue by one place - `@queue[i] = @queue[i+1]`